tests/gradients/parameter_shift/test_parameter_shift.py

```diff
@@ -3462,33 +3461,21 @@ def test_trainable_coeffs(self, mocker, tol, broadcast):
         tapes, fn = qml.gradients.param_shift(tape, broadcast=broadcast)
         # two (broadcasted if broadcast=True) shifts per rotation gate
         # one circuit per trainable H term
-        assert len(tapes) == (2 + 2 if broadcast else 2 * 2 + 2)
-        assert [t.batch_size for t in tapes] == ([2, 2, None, None] if broadcast else [None] * 6)
-        spy.assert_called()
+        assert len(tapes) == (2 if broadcast else 2 * 2)
```

Comment:
Why has the expected number of tapes changed here?

Reply:
We used to go an extra execution per term using `hamiltonian_grad`.  So it we wanted to get the derivative of `expval(2*X(0))`, we would do a bonus measurement of `expval(X(0))`.